pkg/envoy/registry/services_test.go

```diff
@@ -63,13 +70,55 @@ var _ = Describe("Test Proxy-Service mapping", func() {
 
 			expectedSvc2 := service.MeshService{
 				Namespace: tests.Namespace,
-				Name:      svcName2,
+				Name:      fmt.Sprintf("%s.%s", podName, svcName2),
 				Port:      tests.ServicePort,
 				Protocol:  "http",
 			}
+
+			// Subdomain gets called in the ListProxyServices
+			expectedSvc1.Subdomain()
+			expectedSvc2.Subdomain()
 			expectedList := []service.MeshService{expectedSvc1, expectedSvc2}
 
-			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(nil, nil).Times(2)
+			mockKubeController.EXPECT().GetEndpoints(gomock.Any()).Return(&v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: tests.Namespace,
+					Name:      expectedSvc1.Name,
+				},
+				Subsets: []v1.EndpointSubset{
```

Comment:
It looks like `Subsets` doesn't need to be defined at all for this to pass.

Reply:
Similar to https://github.com/openservicemesh/osm/pull/4704#discussion_r877265445. Even though this particular endpoints object doesn't need endpoints, I feel like it raises more questions since the next endpoints object does have IPs